### DISCUSSION TITLE
Would you like some "use of Gatsby" feedback?

### DISCUSSION THREAD

**Discussion by raae** [distributeaid/distributeaid.org#593] [Ideas]
Would you like a Gatsby code review? And if so would you like the feedback as issues or here in the discussion?

**Comment by raae** [comment]
There is a case for moving data from page context to a page query. Especially when more pages are added, as with page context ["Hot reload is taken off the table and the site needs to be rebuilt for changes to reflect." ](https://www.gatsbyjs.com/docs/creating-and-modifying-pages/#incremental-builds-trade-off-of-this-method).

I could refactor from page context to page template for Region pages on my stream on Thursday if that would welcomed?

**Comment by raae** [comment]
Personally I like it when I can read the Gatsby files such as `gatsby-node.js` and get an overview of what is happening. So instead of 

```js
exports.onCreateNode = onCreateNode
```

something like might work:
```js
exports.onCreateNode = (gatsbyUtils) => {
  createRegionNodesFromMarkdown(gatsbyUtils)
  createSubRegionNodesFromMakdwon(gatsbyUtils)
  // and so forth
}
```

**Reply by raae** [reply]
I see you did this in https://github.com/distributeaid/distributeaid.org/commit/bd71d5ae3b9d72b46e6e41171958671c0bfa6a99, that makes me so happy 😍

**Comment by raae** [comment]
Using [the filesystem route api](https://www.gatsbyjs.com/docs/reference/routing/creating-routes/#using-the-file-system-route-api) makes it, in my opinion. easier to navigate the page structure for new developers. Especially since there is already a lot of hard coded pages, and then you can see them side by side.

For instance inside the regions folder, together with `index.tsx` you could have the routes `{DARegion.name}.tsx` and `{DASubregion.region.name}/{DASubregion.name}.tsx`. Not 100% sure on the syntax, but close enough for this purpose of this feedback 🤪

**Comment by jtfairbank** [comment]
Wow @raae this is fantastic!  I'll definitely be going through your suggestions and making issues / implementing them over time.

> I could refactor from page context to page template for Region pages on my stream on Thursday if that would welcomed?

YES PLEASE!  That'd be super cool.  Let me know if you want me or @hola-soy-milk to join, or if you'd like to join us on our OSS Thursdays stream to do this!

**Comment by hola-soy-milk** [comment]
Yay, @raae! Thank you
